sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -275,6 +275,10 @@ private[sql] class JDBCRDD(
    */
   private def compileFilter(f: Filter): String = f match {
     case EqualTo(attr, value) => s"$attr = ${compileValue(value)}"
+    // Since the null-safe equality operator is not a standard SQL operator,
+    // This was written as using is-null and normal equality.
+    case EqualNullSafe(attr, value) =>
+      s"($attr = ${compileValue(value)} OR ($attr IS NULL AND ${compileValue(value)} IS NULL))"
```

Comment:
Yes.. it looks so. There can be a problem if the query uses the result of  comparison `<=>`.

Let me correct this as `(NOT ($attr <> ${compileValue(value)} OR $attr IS NULL OR ${compileValue(value)} IS NULL) OR ($attr IS NULL AND ${compileValue(value)} IS NULL))`.

(BTW, I found `<=>` actually a standard but as `IS NOT DISTINCT FROM` from SQL1999)